lib/shipcloud.rb

```diff
@@ -52,7 +52,7 @@ def self.configure
 
   def self.api_headers
     API_HEADERS.merge(
-      "Affiliate-ID" => configuration.affiliate_id || DEFAULT_AFFILIATE_ID
+      "Affiliate-ID" => configuration.affiliate_id || DEFAULT_AFFILIATE_ID,
```

Comment:
I think that line/merge is redundant now, as the merge is done in [Line 31 in lib/shipcloud/request/connection.rb](https://github.com/shipcloud/shipcloud-ruby/pull/38/files#diff-8b781b79018ef926e2ddc174592a888fR31)? Or am I missing something?

Reply:
We have three possible variants of `affiliate_id`. First is `DEFAULT_AFFILIATE_ID`, second comes from the initialiser configuration, and the third belongs to an entity, which uses shipcloud, and is different from the second one, since different entities use different affiliate IDs under the same initialiser. Here we deal with the two of them, and the third one is covered in the [Line 31 in lib/shipcloud/request/connection.rb](https://github.com/shipcloud/shipcloud-ruby/pull/38/files#diff-8b781b79018ef926e2ddc174592a888fR31).

Reply:
Sorry, I forgot to add additional links to my previous comment.
What I meant was, that the merge is done in [Line 31 in lib/shipcloud/request/connection.rb](https://github.com/shipcloud/shipcloud-ruby/pull/38/files#diff-8b781b79018ef926e2ddc174592a888fR31) (`Shipcloud.api_headers.merge("Affiliate-ID" => affiliate_id`). If I'm not mistaken the given `affiliate_id` in this merge is either the "per Request `affiliate_id`" or as a Fallback the affiliate_id return by the `Shipcloud.affiliate_id` ([this file Line 84](https://github.com/shipcloud/shipcloud-ruby/pull/38/files#diff-cfc978964feed4c0cdb6c9f1e944afb3R84)) Method, which is the same `configuration.affiliate_id || DEFAULT_AFFILIATE_ID` Fallback as in this line here.

So whatever is added with `"Affiliate-ID"` key in this line will be overwritten by [Line 31 in lib/shipcloud/request/connection.rb](https://github.com/shipcloud/shipcloud-ruby/pull/38/files#diff-8b781b79018ef926e2ddc174592a888fR31).

Reply:
I'm not sure about removing [Line 55 from shipcloud.rb](https://github.com/shipcloud/shipcloud-ruby/pull/38#discussion-diff-303446111L55), because if this method is called somewhere without a http request, it will return headers without `Affiliate-ID`.

Reply:
Okay, let's use `Shipcloud.affiliate_id` here for now:

```suggestion
      "Affiliate-ID" => affiliate_id,
```

This should reduce the duplication and I will have look at the default affiliate id handling later on.